server/src/main/java/thesistrack/ls1/service/ApplicationService.java

```diff
@@ -0,0 +1,219 @@
+        this.mailingService = mailingService;
+        this.topicRepository = topicRepository;
+        this.thesisRepository = thesisRepository;
+        this.thesisService = thesisService;
+    }
+
+    public Page<Application> getAll(String searchString, ApplicationState[] states, int page, int limit, String sortBy, String sortOrder) {
+        Sort.Order order = new Sort.Order(sortOrder.equals("asc") ? Sort.Direction.ASC : Sort.Direction.DESC, sortBy);
+
+        return applicationRepository
+                .searchApplications(new HashSet<>(Arrays.stream(states).map(Enum::toString).toList()), searchString.toLowerCase(), PageRequest.of(page, limit, Sort.by(order)));
+    }
+
+    @Transactional
+    public Application createLegacyApplication(
+            LegacyCreateApplicationPayload payload,
+            MultipartFile examinationReport,
+            MultipartFile cv,
+            MultipartFile degreeReport
+    ) {
+        Instant currentTime = Instant.now();
+
+        User student = userRepository.findByUniversityId(payload.getUniversityId()).orElseGet(() -> {
+            User user = new User();
+
+            user.setJoinedAt(currentTime);
+
+            return user;
+        });
+        Application application = new Application();
+
+        student.setUniversityId(payload.getUniversityId());
+        student.setMatriculationNumber(payload.getMatriculationNumber());
+        student.setIsExchangeStudent(payload.getIsExchangeStudent());
+        student.setFirstName(payload.getFirstName());
+        student.setLastName(payload.getLastName());
+        student.setGender(payload.getGender());
+        student.setNationality(payload.getNationality());
+        student.setEmail(payload.getEmail());
+        student.setStudyDegree(payload.getStudyDegree());
+        student.setStudyProgram(payload.getStudyProgram());
+        student.setEnrolledAt(payload.getEnrolledAt());
+        student.setSpecialSkills(payload.getSpecialSkills());
+        student.setInterests(payload.getInterests());
+        student.setProjects(payload.getProjects());
+        student.setResearchAreas(payload.getResearchAreas());
+        student.setFocusTopics(payload.getFocusTopics());
+        student.setUpdatedAt(currentTime);
+
+        student.setExaminationFilename(uploadService.store(examinationReport));
+        student.setCvFilename(uploadService.store(cv));
+
+        if (degreeReport != null && !degreeReport.isEmpty()) {
+            student.setDegreeFilename(uploadService.store(degreeReport));
+        }
+
+        application.setUser(student);
+        application.setThesisTitle(payload.getThesisTitle());
+        application.setMotivation(payload.getMotivation());
+        application.setState(ApplicationState.NOT_ASSESSED);
+        application.setDesiredStartDate(payload.getDesiredStartDate());
+        application.setCreatedAt(currentTime);
+
+        mailingService.sendApplicationCreatedMailToChair(application);
```

Comment:
If I see this correctly, there is no sanitization in place, is it? Neither in the `ApplicationRepository` nor here are values checked and mostly accepted straight out of the payload. This poses a security risk.. Did I miss a sanitization aspect in your system?

Reply:
Down below in `accept()` you perform sanitization ☺️👍

Reply:
I have added more request validation now